pkg/internalapis/csinodetopology/v1alpha1/csinodetopology_types.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CSINodeTopologySpec defines the desired state of CSINodeTopology.
+type CSINodeTopologySpec struct {
+
+	// NodeID refers to the unique ID by which a Node is recognised.
+	NodeID string `json:"nodeID"`
+}
+
+type CRDStatus string
+
+const (
+	// CSINodeTopologyInProgress is used to imply that node topology retrieval is in progress.
+	CSINodeTopologyInProgress CRDStatus = "InProgress"
+	// CSINodeTopologySuccess is used to imply that node topology retrieval is successful.
+	CSINodeTopologySuccess CRDStatus = "Success"
+	// CSINodeTopologyError is used to imply that node topology retrieval resulted in an error.
+	CSINodeTopologyError CRDStatus = "Error"
+)
+
+// CSINodeTopologyStatus defines the observed state of CSINodeTopology.
+type CSINodeTopologyStatus struct {
+	// Status can have the following values: `InProgress`, `Success`, `Error`
+	Status CRDStatus `json:"status"`
```

Comment:
Shouldn't this be omitempty as well?

Reply:
I am planning to mark the `Status` as `InProgress` when i create the CR in `NodeGetInfo` API call. On each reconcile attempt, the status will either be `Success` or `Error`. That is why this is not omitempty. Does it make sense? If not, I can remove the `InProgress` state completely.

Reply:
As per our offline discussion, I removed `InProgress` status and just retained `Success`, `Error`